Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts

```diff
@@ -73,34 +70,116 @@ export const diagnosticsSelector = selectorFamily({
       recognizers,
       crossTrainConfig,
     };
+  },
+});
+
+export const botDiagnosticsSelectorFamily = selectorFamily({
+  key: 'botDiagnosticsSelectorFamily',
+  get: (projectId: string) => ({ get }) => {
+    const botAssets = get(botAssetsSelectFamily(projectId));
+    if (botAssets === null) return [];
 
+    const rootProjectId = get(rootBotProjectIdSelector) ?? projectId;
+    const diagnostics = get(botDiagnosticsState(projectId));
     const diagnosticList: DiagnosticInfo[] = [];
+
     diagnostics.forEach((d) => {
       diagnosticList.push(new BotDiagnostic(rootProjectId, projectId, '', d.source, d));
     });
-    BotIndexer.validate(botAssets).forEach((d) => {
+
+    //manifest.json
+    //Manifest should exist
+    BotIndexer.checkManifest(botAssets).forEach((d) => {
       diagnosticList.push(new BotDiagnostic(rootProjectId, projectId, '', d.source, d));
     });
+
+    return diagnosticList;
+  },
+});
+
+/**
+ * Check bot skill & setting
+ * 1. used skill not existed in setting
+ * 2. appsettings.json Microsoft App Id or Skill Host Endpoint are empty
+ */
+export const skillDiagnosticsSelectorFamily = selectorFamily({
+  key: 'skillDiagnosticsSelectorFamily',
+  get: (projectId: string) => ({ get }) => {
```

Comment:
skillId is better

Reply:
Sorry @hatpick the naming confused you here. In fact, this selector family is used to collect the diagnostics about every bot's connected skill settings. I have updated naming